realm-annotations-processor/src/main/java/io/realm/processor/RealmVersionChecker.java

```diff
@@ -80,7 +81,11 @@ private String checkLatestVersion() {
             conn.setConnectTimeout(CONNECT_TIMEOUT);
             conn.setReadTimeout(READ_TIMEOUT);
             BufferedReader rd = new BufferedReader(new InputStreamReader(conn.getInputStream()));
-            result = rd.readLine();
+            String latestVersion = rd.readLine();
+            // if obtained string does not match the pattern, we are in the separate network.
```

Comment:
if *the* obtained string does not match the pattern, we are in *a* separate network.

Also, i'm not entirely sure what a "separate network" mean? You mean through a proxy or a isolated/private network that might change the response?